File: Labs/Lab2/validAnagram.java
import java.util.*;

public class validAnagram
{
    public boolean isAnagram(String s, String t) 
    {
        HashMap<Character, Integer> sCount = counter(s);
        HashMap<Character, Integer> tCount = counter(t);
        return sCount.equals(tCount);
    }
    
    public static HashMap<Character, Integer> counter(String s) {
        HashMap<Character, Integer> count = new HashMap<>();
        for (int i = 0; i < s.length(); i++)
        {
            if (count.containsKey(s.charAt(i)))
            {
                count.put(s.charAt(i), count.get(s.charAt(i)) + 1);
            }
            else
            {
                count.put(s.charAt(i), 1);
            }
        }
        return count;
    }
}

File: Labs/Lab2/validAnagram.py
class validAnagram(object)
    def isAnagram(self, s, t):
        return self.counter(s) == self.counter(t)
        
    def counter(self, s):
        count = {}
        for c in s:
            if c in count:
                count[c] +=1
            else:
                count[c] = 1
        return count
